src/api/streams/updateStream.js

```diff
@@ -1,21 +1,51 @@
 /* @flow strict-local */
 import type { ApiResponse, Auth } from '../transportTypes';
+import type { Stream } from '../modelTypes';
 import { apiPatch } from '../apiFetch';
 
 /**
  * https://zulip.com/api/update-stream
  *
- * NB the caller must adapt for old-server compatibility; see comment.
+ * NB the caller must adapt for old-server compatibility; see comments.
  */
+// Current to FL 121; property ordering follows the doc.
 // TODO(#4659): Once we pass the feature level to API methods, this one
-//   should encapsulate a switch at feature level 64.  See its call sites.
-// TODO(server-4.0): Simplify that away.
+//   should encapsulate various feature-level switches; see comments.
 export default (
   auth: Auth,
   id: number,
-  property: string,
-  value: string | boolean,
-): Promise<ApiResponse> =>
-  apiPatch(auth, `streams/${id}`, {
-    [property]: value,
-  });
+  params: $ReadOnly<{|
+    // TODO(server-4.0): For these two, it's no longer acceptable to give
+    //   the server values without `JSON.stringify`ing them first. There was
+    //   a server API change that was accidentally incompatible:
+    //     https://github.com/zulip/zulip-mobile/issues/4747#issuecomment-946362729
+    //     https://github.com/zulip/zulip-mobile/pull/4748#issuecomment-852254404
+    description?: $PropertyType<Stream, 'description'>,
+    new_name?: $PropertyType<Stream, 'name'>,
```

Comment:
I think perhaps the simplest solution for this comment is to say:
```suggestion
    // TODO(#4659): Once we pass the feature level to API methods, this one
    //   should encapsulate a switch at FL 64 related to these two parameters.
    //   See call sites.
    description?: $PropertyType<Stream, 'description'>,
    new_name?: $PropertyType<Stream, 'name'>,
```

and delete the TODO-4659 comment on the method as a whole -- i.e. effectively move it here -- and then leave the comment at the call site unchanged.

Then for other things that should be encapsulated, a one-line comment like
`// TODO(#4659): Encapsulate this.`
can suffice, because the TODO-server comments already explain what changed between versions. I think that means `is_announcement_only` and `'forever'`?